src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -109,6 +116,8 @@ public SamlSecurityRealm(String signOnUrl, String idpMetadata, String displayNam
     if (usernameCaseConversion != null && !usernameCaseConversion.isEmpty()) {
       this.usernameCaseConversion = Util.fixEmptyAndTrim(usernameCaseConversion);
     }
+
+    this.userIdStrategy = userIdStrategy == null ? IdStrategy.CASE_INSENSITIVE : userIdStrategy;
```

Comment:
is the forcing of case required?.
the default should come from the descriptor if `null` (to match the jelly).

The value possibly wants to be set depending on if the user has previously chosen a `usernameCaseConversion` of uppercase or lowercase.